src/openrct2/drawing/NewDrawing.cpp

```diff
@@ -212,34 +212,34 @@ void gfx_draw_line(rct_drawpixelinfo* dpi, int32_t x1, int32_t y1, int32_t x2, i
 }
 
 void gfx_draw_dashed_line(
-    rct_drawpixelinfo* dpi, const int32_t x1, const int32_t y1, const int32_t x2, const int32_t y2,
-    const int32_t dashedLineSegmentLength, const int32_t colour)
+    rct_drawpixelinfo* dpi, const ScreenLine& screenLine, const int32_t dashedLineSegmentLenght, const int32_t color)
 {
-    assert(dashedLineSegmentLength > 0);
+    assert(dashedLineSegmentLenght > 0);
 
     const auto drawingEngine = dpi->DrawingEngine;
     if (drawingEngine != nullptr)
     {
         constexpr int32_t precisionFactor = 1000;
 
-        const int32_t dashedLineLength = std::hypot(x2 - x1, y2 - y1);
-        const int32_t lineSegmentCount = dashedLineLength / dashedLineSegmentLength / 2;
+        const int32_t dashedLineLenght = std::hypot(
```

Comment:
```suggestion
        const int32_t dashedLineLength = std::hypot(
```
It's "length" not "lenght"